automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/terraredirect/RedirectSpec.scala

```diff
@@ -0,0 +1,86 @@
+import org.broadinstitute.dsde.firecloud.page.methodrepo.MethodRepoPage
+import org.broadinstitute.dsde.workbench.config.UserPool
+import org.broadinstitute.dsde.workbench.fixture.{BillingFixtures, TestReporterFixture, WorkspaceFixtures}
+import org.broadinstitute.dsde.workbench.service.test.{CleanUp, WebBrowserSpec}
+import org.scalatest.{BeforeAndAfterAll, FreeSpec, Matchers}
+
+
+/**
+  * This spec asserts that a user who loads specific pages in the FireCloud UI is appropriately
+  * redirected to the Terra UI.
+  *
+  * As of this writing, redirects are disabled, because the redirects will break OTHER tests.
+  * To verify THIS test passes, you must do two things:
+  *   1) unignore any test you want to run, inside this file
+  *   2) set the UI's flag to enable redirects. You can do this by either or:
+  *     a) changing "terraRedirectsEnabled" to true in the UI instance's config.json
+  *     b) in the UI's config.cljs, change the terra-redirects-enabled function to return a hardcoded
+  *         value of true instead of returning (get @config "terraRedirectsEnabled" false)
+  */
+class RedirectSpec extends FreeSpec with BeforeAndAfterAll with Matchers with WebBrowserSpec
+  with UserFixtures with WorkspaceFixtures with BillingFixtures with CleanUp with LazyLogging with TestReporterFixture {
+
+  val user = UserPool.chooseAnyUser
+
+  // TODO: this test sends traffic from a FiaB or other test environment directly to the Terra dev server.
+  //  tests will fail if the dev server is unresponsive, and tests will cause additional load on the dev server.
+  //  is this acceptable?
+  private def isTerraUrl(url: String) = {
+    // should this use an environment-specific full hostname instead of contains()?
+    url.contains("terra.bio") ||
+    (url.contains("saturn") && url.contains("appspot.com"))
+  }
+
+  // negative test - ensure that Library/Methods Repo/whatever does NOT redirect
+  "A user who visits FireCloud should NOT be redirected to the same location in Terra for" - {
+
+    "methods repo" in {
+      implicit val authToken = user.makeAuthToken()
+      withWebDriver { implicit driver =>
+        withSignIn(user) { _ =>
+          val methodRepoPage = new MethodRepoPage().open
+          // give the page a chance to issue a redirect, if a redirect mistakenly existed on this page
+          Thread.sleep(1000)
+          // are we still on the methods repo page?
+          await ready methodRepoPage.methodRepoTable
+        }
+      }
+    }
+
+  }
+
+
+  "A user who visits FireCloud should be redirected to the same location in Terra for" - {
+
+    "workspace summary page" ignore {
+      implicit val authToken = user.makeAuthToken()
+      withCleanBillingProject(user) { billingProject =>
+        withWorkspace(billingProject, "Terra_redirect_spec") { workspaceName =>
+          withWebDriver { implicit driver =>
+            // TODO: withSignIn will throw an error at the end of its closure, when it tries to sign out the user.
```

Comment:
I'm pretty sure the main purpose of signing-out from `withSignIn` (other than the very few tests that need to sign in multiple times) was a shot-in-the-dark attempt to help us from being recognized as a robot by Google. If we want to eliminate the error from trying to sign out, we could extract a `signIn` function from https://github.com/broadinstitute/firecloud-ui/blob/develop/automation/src/test/scala/org/broadinstitute/dsde/firecloud/fixture/UserFixtures.scala#L73 and use that instead of `withSignIn`.

Reply:
agreed on extracting the function - I was just hoping to avoid doing that :(

Reply:
What's the consequence of not extracting a function? Does the test fail because of this or is there just an ugly error logged somewhere?

Reply:
this may helps `try page.signOut() catch nonFatalAndLog(s"ERROR logging out user: ${user.email}")`

Reply:
the test doesn't fail, as evidenced by it passing in the second-to-last commit. The test log does contain a stack trace - that's the only consequence I can see - I'm tossing it out here to ask if anyone else can predict another consequence, or if folks object strongly to the additional stacktrace. (for context, remember that these tests are being sunset along with the UI and we'll be paying less attention to them)